drivers/pinmux/stm32/pinmux_stm32.c

```diff
@@ -31,80 +31,25 @@
 #endif /* CONFIG_SOC_SERIES_STM32MP1X */
 /* base address for where GPIO registers start */
 #define GPIO_PORTS_BASE       (GPIOA_BASE)
-
-static const uint32_t ports_enable[STM32_PORTS_MAX] = {
-	STM32_PERIPH_GPIOA,
-	STM32_PERIPH_GPIOB,
-	STM32_PERIPH_GPIOC,
-#ifdef GPIOD_BASE
-	STM32_PERIPH_GPIOD,
-#else
-	STM32_PORT_NOT_AVAILABLE,
-#endif
-#ifdef GPIOE_BASE
-	STM32_PERIPH_GPIOE,
-#else
-	STM32_PORT_NOT_AVAILABLE,
-#endif
-#ifdef GPIOF_BASE
-	STM32_PERIPH_GPIOF,
-#else
-	STM32_PORT_NOT_AVAILABLE,
-#endif
-#ifdef GPIOG_BASE
-	STM32_PERIPH_GPIOG,
-#else
-	STM32_PORT_NOT_AVAILABLE,
-#endif
-#ifdef GPIOH_BASE
-	STM32_PERIPH_GPIOH,
-#else
-	STM32_PORT_NOT_AVAILABLE,
-#endif
-#ifdef GPIOI_BASE
-	STM32_PERIPH_GPIOI,
-#else
-	STM32_PORT_NOT_AVAILABLE,
-#endif
-#ifdef GPIOJ_BASE
-	STM32_PERIPH_GPIOJ,
-#else
-	STM32_PORT_NOT_AVAILABLE,
-#endif
-#ifdef GPIOK_BASE
-	STM32_PERIPH_GPIOK,
-#else
-	STM32_PORT_NOT_AVAILABLE,
-#endif
+#define GPIO_DEVICE(gpio_port)						\
+	COND_CODE_1(DT_NODE_HAS_STATUS(DT_NODELABEL(gpio_port), okay),	\
+		    (DEVICE_DT_GET(DT_NODELABEL(gpio_port))),		\
+		    ((const struct device *)NULL))
+
+const struct device *gpio_ports[STM32_PORTS_MAX] = {
```

Comment:
This defines an array of constant pointers, but we could make the array itself constant and move it from RAM to FLASH.
```
const struct device * const gpio_ports[STM32_PORTS_MAX] = {
```

Reply:
This saves a few bytes indeed, txs!